python/tvm/relax/frontend/onnx_frontend.py

```diff
@@ -0,0 +1,1158 @@
+        relax.BlockBuilder._current = prev_bb
+
+        # This dict is used by the Mapper mutator to replace the globar vars
+        # in the relax_mod with global_vars registered with the in-use block builder.
+        global_var_dict = {}
+        for gv, func in relax_mod.functions.items():
+            if gv.name_hint != "main":
+                global_var_dict[gv] = self.bb.add_func(func, gv.name_hint)
+
+        # This dict is used by the Mapper mutator to replace the relax vars
+        # with the inputs.
+        relax_input_dict = {}
+        for relax_var in relax_inputs:
+            if isinstance(relax_var, relax.Var):
+                relax_input_dict[relax_var.name_hint] = relax_var
+
+        @relax.expr_functor.mutator
+        class Mapper(PyExprMutator):
+            def visit_var_(self, var_node: Var):
+                if var_node.name_hint in relax_input_dict:
+                    return relax_input_dict[var_node.name_hint]
+                return var_node
+
+            def visit_global_var_(self, gv_node: GlobalVar):
+                if gv_node in global_var_dict:
+                    return global_var_dict[gv_node]
+                return gv_node
+
+        updated_func = Mapper().visit_expr(relax_mod["main"])
+
+        var_bindings = updated_func.body.blocks[0].bindings
+        if isinstance(updated_func.ret_struct_info, relax.TupleStructInfo):
+            # Returning a tuple.
+            final_binding = var_bindings[-2]
+            for binding in var_bindings[:-2]:
+                self.bb.emit_normalized(binding)
+        else:
+            final_binding = var_bindings[-1]
+            for binding in var_bindings[:-1]:
+                self.bb.emit_normalized(binding)
+
+        return final_binding.value
+
+    def _convert_operator(self, op_name, inputs, attrs, opset):
+        """Convert ONNX operator into a Relax operator.
+        The converter must specify conversions explicitly for incompatible name, and
+        apply handlers to operator attributes.
+        Parameters
+        ----------
+        op_name : str
+            Operator name, such as Convolution, FullyConnected
+        inputs : list of tvm.relax.function.Function
+            List of inputs.
+        attrs : dict
+            Dict of operator attributes
+        opset : int
+            Opset version
+        Returns
+        -------
+        sym : tvm.relax.function.Function
+            Converted relax function
+        """
+        convert_map = _get_convert_map(opset)
+        if op_name in convert_map:
+            convert_class = convert_map[op_name]
+            op_function = convert_class.get_converter(opset)
+            # If the op_function is a subclass of Relay OnnxOpConverter then it is a relay op.
+            if issubclass(convert_class, RelayOnnxOpConverter):
+                relay_inputs = self._relay_input_adapter(inputs)
+                # The op_function might change the inputs to the relay op. Use a copy of the inputs.
```

Comment:
In what sense would the inputs be changed? The AST being transformed?

Reply:
I've updated the comment --- the Relay op might assign different elements to the input array